langchain-core/src/callbacks/manager.ts

```diff
@@ -1252,6 +1254,32 @@ export class CallbackManager
         callbackManager.addMetadata(localMetadata ?? {}, false);
       }
     }
+
+    for (const {
+      contextVar,
+      inheritable,
+      handlerClass,
+      envVar,
+    } of getConfigureHooks()) {
+      const createOne =
+        envVar && getEnvironmentVariable(envVar) === "true" && handlerClass;
+      if (getContextVariable(contextVar) || createOne) {
+        const varHandler = ensureHandler(
+          (getContextVariable(contextVar) as
```

Comment:
nit: Would prefer a type guard here for `BaseCallbackHandler ` instead of a cast